test/test_constraint.jl

```diff
@@ -1684,4 +1684,93 @@ function test_eltype_for_constraint_primal_complex_float64()
     return
 end
 
+function test_tuple_shape_vector()
+    model = Model()
+    @variable(model, t)
+    @variable(model, x[1:3])
+    @variable(model, u)
+    @variable(model, v)
+    @variable(model, w)
+    for set in (
+        MOI.ExponentialCone(),
+        MOI.DualExponentialCone(),
+        MOI.PowerCone(0.5),
+        MOI.DualPowerCone(0.5),
+    )
+        c = @constraint(model, (u, v, w) in set)
+        obj = constraint_object(c)
+        @test obj.func == [u, v, w]
+        @test reshape_vector(obj.func, obj.shape) == (u, v, w)
+        @test reshape_set(obj.set, obj.shape) == obj.set
+    end
+    for set in
+        (MOI.SecondOrderCone(4), SecondOrderCone(), MOI.GeometricMeanCone(4))
+        c = @constraint(model, (t, x) in set)
```

Comment:
This is now a generic implementation that works for arbitrary cones. If it has a downside, it's that it's too permissive. Effectively, the tuple vectorized by concatenating the elements, and doesn't concern itself with whether the structure of the tuple makes sense for the set. So these three constraints are all valid:
```julia
@constraint(model, (t, x) in SecondOrderCone())
@constraint(model, (t, x[1], x[2], x[3]) in SecondOrderCone())
@constraint(model, (t, x[1], x[2:3]) in SecondOrderCone())
```
We don't have a way of saying that `SecondOrderCone` is an epigraph with `Scalar,Vector`, and that `RotatedSecondOrderCone` is a perspective epigraph with `Scalar,Scalar,Vector`. We just know the full dimension.